tests/integration/conftest.py

```diff
@@ -602,3 +602,74 @@ def create_file_structure_function(get_files_list):
     yield
 
     delete_file_structure(get_files_list)
+
+
+@pytest.fixture(scope='module')
+def daemons_handler(get_configuration, request):
+    """Handler of Wazuh daemons.
+
+    It uses `daemons_handler_configuration` of each module in order to configure the behavior of the fixture.
+    The  `daemons_handler_configuration` should be a dictionary with the following keys:
+        daemons (list, optional): List with every daemon to be used by the module. In case of empty a ValueError
+            will be raised
+        wazuh_control (boolean): Configure to restart all wazuh services. Default `False`.
+        ignore_errors (boolean): Configure if errors in daemon handling should be ignore. This option is available
+        in order to use this fixture along with invalid configuration. Default `False`
+
+    Args:
+        get_configuration (fixture): Gets the current configuration of the test.
+        request (fixture): Provide information on the executing test function.
+    """
+    daemons = []
+    ignore_errors = False
+    wazuh_control = False
+
+    try:
+        daemons_handler_configuration = getattr(request.module, 'daemons_handler_configuration')
+        if 'daemons' in daemons_handler_configuration and not wazuh_control:
+            daemons = daemons_handler_configuration['daemons']
+            if not daemons:
+                logger.error('Daemons Handler: Daemons list is not set')
+                raise ValueError
+
+        if 'wazuh_control' in daemons_handler_configuration:
+            logger.debug(f"Daemons Handler: Wazuh control set to {daemons_handler_configuration['wazuh_control']}")
+            wazuh_control = daemons_handler_configuration['wazuh_control']
+
+        if 'ignore_errors' in daemons_handler_configuration:
+            logger.debug(f"Daemons Handler: Ignore error set to {daemons_handler_configuration['ignore_errors']}")
+            ignore_errors = daemons_handler_configuration['ignore_errors']
+
+    except AttributeError as daemon_configuration_not_set:
+        logger.error('Daemons Handler: Error - daemons_handler_configuration is not set')
+        raise daemon_configuration_not_set
+
+    try:
+        if wazuh_control:
+            logger.debug('Daemons Handler: Debug - Restarting wazuh using wazuh-control')
+            # Restart daemon instead of starting due to legacy used fixture in the test suite.
+            control_service('restart')
+        else:
+            for daemon in daemons:
+                logger.debug(f"Daemons Handler: Restarting {daemon}")
+                # Restart daemon instead of starting due to legacy used fixture in the test suite.
+                control_service('restart', daemon=daemon)
```

Comment:
What happens if we have entered a daemon name that does not exist or is invalid?

Reply:
In that case, the test will raise an error at this `contol_service` line

https://github.com/wazuh/wazuh-qa/blob/f32574624734736870596be059d88722e938ba85/deps/wazuh_testing/wazuh_testing/tools/services.py#L126

Maybe this function needs a refactor in which we can include more verbose and daemons values handling.

Reply:
Done  52b1d67